doc/source/cluster/config.rst

```diff
@@ -86,59 +290,772 @@ If Docker is not installed, add the following commands to ``initialization_comma
+
+* **Required:** No
+* **Importance:** Low
+* **Type:** String
+
+.. _cluster-configuration-container-name:
+
+``docker.container_name``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The name to use when starting the Docker container.
+
+* **Required:** Yes (If Docker is in use.)
+* **Importance:** Low
+* **Type:** String
+* **Default:** ray_container
+
+.. _cluster-configuration-pull-before-run:
+
+``docker.pull_before_run``
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, the latest version of image will be pulled when starting Docker. If disabled, ``docker run`` will only pull the image if no cached version is present.
+
+* **Required:** No
+* **Importance:** Medium
+* **Type:** Boolean
+* **Default:** ``True``
+
+.. _cluster-configuration-run-options:
+
+``docker.run_options``
+~~~~~~~~~~~~~~~~~~~~~~
+
+The extra options to pass to ``docker run``.
+
+* **Required:** No
+* **Importance:** Medium
+* **Type:** List of String
+* **Default:** ``[]``
+
+.. _cluster-configuration-head-run-options:
+
+``docker.head_run_options``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The extra options to pass to ``docker run`` for head node only.
+
+* **Required:** No
+* **Importance:** Low
+* **Type:** List of String
+* **Default:** ``[]``
+
+.. _cluster-configuration-worker-run-options:
+
+``docker.worker_run_options``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The extra options to pass to ``docker run`` for worker nodes only.
+
+* **Required:** No
+* **Importance:** Low
+* **Type:** List of String
+* **Default:** ``[]``
+
+.. _cluster-configuration-disable-automatic-runtime-detection:
+
+``docker.disable_automatic_runtime_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not try to use the NVIDIA Container Runtime if GPUs are present.
+
+* **Required:** No
+* **Importance:** Low
+* **Type:** Boolean
+* **Default:** ``False``
+
+
+.. _cluster-configuration-disable-shm-size-detection:
+
+``docker.disable_shm_size_detection``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If enabled, Ray will not automatically specify the size ``/dev/shm`` for the started container and the runtime's default value (64MiB for Docker) will be used.
+
+* **Required:** No
+* **Importance:** Low
+* **Type:** Boolean
+* **Default:** ``False``
+
+
+.. _cluster-configuration-ssh-user:
+
+``auth.ssh_user``
+~~~~~~~~~~~~~~~~~
+
+The user that Ray will authenticate with when launching new nodes.
+
+* **Required:** Yes
+* **Importance:** High
+* **Type:** String
+
+.. _cluster-configuration-ssh-private-key:
+
+``auth.ssh_private_key``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. tabs::
+    .. group-tab:: AWS
+
+        The path to an existing private key for Ray to use. If not configured, Ray will create a new private keypair (default behavior). If configured, the key must be added to the project-wide metadata and ``KeyName`` has to be defined in the :ref:`node configuration <cluster-configuration-node-config>`.
```

Comment:
If someone knows the straightforward behavior here please comment. I can check it later if not.

Reply:
In order to access an AWS instance, EC2 requires a KeyPair. By default, Ray will create one called `ray-autoscaler_{key #}_{region}`. The private key is stored at `~/.ssh/ray-autoscaler_{key #}_{region}.pem`. To specify your own key, you need to specify your AWS KeyPair name, and the corresponding location of the private key. 

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html

Key # is an incrementing field. ray puts an arbitrary limit of 30 keys.

Reply:
^NOTE that the 30 keys is per AWS account not per laptop or what-not.